### PR TITLE
Post: mlr3keras Boston Housing Regression

### PR DESCRIPTION
**Pull request by henrifnk**
This is an example on how to use `mlr3keras`.
The Post relies on [RKeras Basic Regression Tutorial](https://keras.rstudio.com/articles/tutorial_basic_regression.html).

### CODE REVIEW

File: _posts/2020-05-27-mlr3keras-boston-housing-task/mlr3keras-boston-housing-task.Rmd [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+library(mlr3)
+library(mlr3keras)
+library(mlr3pipelines)
+library(keras)
+library(ggplot2)
+```
+
+# Basic Regression Task - Boston Housing Prices
+
+In this use case we build a model that predicts the Median value of owner-occupied homes in Boston in $1000's (`medv`).
```
Comment (pfistfl) [inline _posts/2020-05-27-mlr3keras-boston-housing-task/mlr3keras-boston-housing-task.Rmd]: case,

File: _posts/2020-05-27-mlr3keras-boston-housing-task/mlr3keras-boston-housing-task.Rmd [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+
+learner$param_set$values$epochs = 500L
+learner$param_set$values$model = build_model()
+learner$param_set$values$validation_split = 0.2
+```
+
+### Callbacks
+
+Another hypereparameter is `callback`.
+A `callback` is a set of functions, bundled in a list that can be used to explore specified progress stages during training.
```
Comment (pfistfl) [inline _posts/2020-05-27-mlr3keras-boston-housing-task/mlr3keras-boston-housing-task.Rmd]: explore and manipulate.

File: _posts/2020-05-27-mlr3keras-boston-housing-task/mlr3keras-boston-housing-task.Rmd [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+Accordingly a defined `callback` function is called at its' specified stage of training to monitor or display specified model parameters at this stage.
+The history callback for example is applied to each training progress of a model by default.
```
Comment (pfistfl) [inline _posts/2020-05-27-mlr3keras-boston-housing-task/mlr3keras-boston-housing-task.Rmd]: The history callback, for example, tracks the train and validation loss after each epoch.

File: _posts/2020-05-27-mlr3keras-boston-housing-task/mlr3keras-boston-housing-task.Rmd [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+It stores the training history in a `history` slot of its' concerning model object.
+For a list of all implemented callbacks see [Keras Callbacks](https://keras.io/callbacks/).
+
+In our first learner we want to reduce information about training progress to a single dot printed out at the end of each applied epoch.
+Therefore we create a `LambdaCallback` that is called at each epochs end and assign it to the learners' `callbacks`-hyperparameter.
+
```
Comment (pfistfl) [inline _posts/2020-05-27-mlr3keras-boston-housing-task/mlr3keras-boston-housing-task.Rmd:183]: Explain what this does, i.e. print a "." after every epoch and newline after every 80 epochs

File: _posts/2020-05-27-mlr3keras-boston-housing-task/mlr3keras-boston-housing-task.Rmd [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+    if (epoch %% 80 == 0) cat("\n")
+    cat(".")
+  }
+) 
+
+learner$param_set$values$callbacks = list(print_dot_callback)
+```
+
+Callback can also be called through `r gh_pkg("mlr-org/mlr3keras")` syntax.
+As not all callbacks are implemented yet [mlr3keras callbacks](https://github.com/mlr-org/mlr3keras/blob/master/R/keras_callbacks.R) is still in progress.
```
Comment (pfistfl) [inline _posts/2020-05-27-mlr3keras-boston-housing-task/mlr3keras-boston-housing-task.Rmd]: `mlr3keras` contains a set of often-used callbacks for the user's convenience.

File: _posts/2020-05-27-mlr3keras-boston-housing-task/mlr3keras-boston-housing-task.Rmd [hunk 6]
```diff
@@ -0,0 +1,332 @@
...
+* `cb_tb`: Tensorboard callback
+* `cb_lr_log`: Learning rate logger callback
+* `LogMetrics`: Batch-wise Metrics Logger Callback
+* `SetLogLR`: Batch-wise Metrics Setter Callback
+
+Initialize early stopping like this:
+
+```{r mlr3cb}
+# early_stop <- callback_early_stopping(monitor = "val_loss", patience = 20) # keras syntax
+early_stop <- cb_es(monitor = "val_loss", patience = 20) # mlr3 syntax
```
Comment (pfistfl) [inline _posts/2020-05-27-mlr3keras-boston-housing-task/mlr3keras-boston-housing-task.Rmd]: We mix up `<-` und `=` throughout the article, please choose one and be consistent

File: _posts/2020-05-27-mlr3keras-boston-housing-task/mlr3keras-boston-housing-task.Rmd [hunk 7]
```diff
@@ -0,0 +1,332 @@
...
+paste0(
+  "The models average error is at ",
+  round(predict_boston$score(msr("regr.mae")) * 1000, 2),
+  "$")
+```
+
+
+# Set Seed
+
+`Keras R Studio` is a system running code on multiple platforms.
```
Comment (pfistfl) [inline _posts/2020-05-27-mlr3keras-boston-housing-task/mlr3keras-boston-housing-task.Rmd]: I don't like this sentence. Instead:
`Keras` can run code across multiple devices.

Comment (pfistfl) [reply _posts/2020-05-27-mlr3keras-boston-housing-task/mlr3keras-boston-housing-task.Rmd]: Write one sentence about sources of stochasticity: 
- initial weights 
- ordering in batches
- stochasticity in model architecture
- parallelism
True eproducibility can only be guaranteed if code is not run in parallel and initial weights are fixed.

**Comment by pfistfl** [comment]
subsumed by #89
